libs/dal/src/repositories/base-repository.ts

```diff
@@ -113,6 +113,12 @@ export class BaseRepository<T_Query, T_Response> {
     };
   }
 
+  async upsertMany(data: T_Query[]) {
+    const promises = data.map((entry) => this.MongooseModel.findOneAndUpdate(entry, entry, { upsert: true }));
+
+    return await Promise.all(promises);
+  }
```

Comment:
The other option recommend was to use `bulkWrite` but I found it harder to make it work with all the changes I had to do. Maybe we can review this and its performance in the near future.

Reply:
This is totally fine for now, can re evelaute later.